src/ipc/Kid.cc

```diff
@@ -57,15 +62,39 @@ Kid::stop(PidStatus const theExitStatus)
     assert(startTime != 0);
 
     isRunning = false;
+    stopTime = squid_curtime;
+    status = theExitStatus;
 
-    time_t stop_time;
-    time(&stop_time);
-    if ((stop_time - startTime) < fastFailureTimeLimit)
+    if ((stopTime - startTime) < fastFailureTimeLimit)
         ++badFailures;
     else
         badFailures = 0; // the failures are not "frequent" [any more]
 
-    status = theExitStatus;
+    reportStopped(); // after all state changes
+}
+
+/// describes a recently stopped kid
+void
+Kid::reportStopped() const
```

Comment:
I am not against making this 'debugs' or adding some 'debugs', however there may be many other 'syslog' calls in master code needing similar change? I would do all that changes in a separate issue. IIRC, we were also thinking about using debugs everywhere and redirecting messages to syslog by _db_set_syslog() method.